smithy-typescript-codegen/src/main/java/software/amazon/smithy/typescript/codegen/integration/HttpBindingProtocolGenerator.java

```diff
@@ -214,6 +216,72 @@ public void generateResponseSerializers(GenerationContext context) {
         }
     }
 
+    @Override
+    public void generateRouter(GenerationContext context) {
+        TopDownIndex topDownIndex = TopDownIndex.of(context.getModel());
+
+        context.getWriter().addImport("restjson", null, "@aws-smithy/server-common");
+
+        context.getWriter().openBlock("export const $LRouter = new $L([", "]);",
+                StringUtils.uncapitalize(context.getService().getId().getName()),
+                "restjson.Router",
+                () -> {
+                    Set<OperationShape> containedOperations = new TreeSet<>(
+                            topDownIndex.getContainedOperations(context.getService()));
+
+                    for (OperationShape operation : containedOperations) {
+                        OptionalUtils.ifPresentOrElse(
+                                operation.getTrait(HttpTrait.class),
+                                httpTrait -> generateUriSpec(context, operation, httpTrait),
+                                () -> LOGGER.warning(String.format(
+                                        "Unable to generate %s uri spec for %s because it does not have an "
+                                                + "http binding trait", getName(), operation.getId())));
+
+                    }
+               }
+       );
+    }
+
+    private void generateUriSpec(GenerationContext context,
+                                 OperationShape operation,
+                                 HttpTrait httpTrait) {
+        context.getWriter().openBlock("new restjson.UriSpec(", "),", () -> {
```

Comment:
nit: you're calling `context.getWriter()` quite a bit. Calling it once and storing it would save some space